src/internal/operators/combineAll.ts

```diff
@@ -53,6 +56,10 @@ export function combineAll<R>(project: (...values: Array<any>) => R): OperatorFu
  * @return {Observable<T>}
  * @name combineAll
  */
-export function combineAll<T, R>(project?: (...values: Array<any>) => R): OperatorFunction<T, R> {
-  return (source: Observable<T>) => lift(source, new CombineLatestOperator(project, null));
+export function combineAll<T, R>(project?: (...values: Array<any>) => R): OperatorFunction<ObservableInput<T>, R|T[]> {
+  return (source: Observable<ObservableInput<T>>) => source.pipe(
+    toArray(),
+    concatMap((sources) => combineLatest(sources)),
+    project ? map((args) => args.length === 1 ? project(args) : project(...args)) : identity as any
```

Comment:
question: can `mapOneOrManyArgs` be used here?

Reply:
Yup! Good catch.